src/engines/preprocessing_engine.py

```diff
@@ -20,7 +21,7 @@ def get_columns(self):
         return self.state.imported_data.columns
 
     def set_state(self, columns):
-        self.state.imported_data = self.state.raw_data[columns].copy()
+        self.state.imported_data = self.state.imported_data[columns]
```

Comment:
I am not sure that we should change it

Reply:
After reduction, columns of imported data change names, so if the user then wants to reject some columns from the widget next to the reduction widget, then he will choose some subset of new columns, while raw_data still has old column labels. In this case the app will crash, but I can rethink how to manage it in another way.

Reply:
![image](https://user-images.githubusercontent.com/81808072/197642924-56af4485-35be-44b8-a621-e27daa51ab66.png)

Reverted, because raw_works as you suggested. Btw do we want to achieve a left or right result assuming that:

- df1 is raw_data and df_2 is imported_data,
- we can modify imported_data later?